homeassistant/components/sonos/speaker.py

```diff
@@ -202,8 +193,11 @@ def setup(self, entry: ConfigEntry) -> None:
                 self.hass, SONOS_CREATE_AUDIO_FORMAT_SENSOR, self, audio_format
             )
 
-        if battery_info := fetch_battery_info_or_none(self.soco):
-            self.battery_info = battery_info
+        try:
+            self.battery_info = self.fetch_battery_info()
+        except SonosUpdateError:
```

Comment:
How does this raise `SonosUpdateError`? https://github.com/jjlawren/home-assistant/blob/784be5893181764d7414aa2a21ea1feb2653f661/homeassistant/components/sonos/exception.py#L10

Reply:
That method is wrapped in `@soco_error` which catches several exception types and then raises `SonosUpdateError`.

Reply:
🤦 forgot about the decorator... always forget about the decorators

👍 https://github.com/jjlawren/home-assistant/blob/784be5893181764d7414aa2a21ea1feb2653f661/homeassistant/components/sonos/helpers.py#L83